opal/util/net.c

```diff
@@ -405,7 +403,6 @@ opal_net_get_hostname(const struct sockaddr *addr)
        int err = errno;
        opal_output (0, "opal_sockaddr2str failed:%s (return code %i)\n",
                     gai_strerror(err), error);
-       free (name);
```

Comment:
Because I was curious why we release this buffer in all error cases but never on the successful path, so I went read the code of the get_hostname_buffer function. The returned buffer is associated with the calling thread (TLS), and we better don't free it at all.